### PR TITLE
Merge from Upstream Blockly

### PR DESCRIPTION
**Pull request by thisandagain**
- [x] Resolve outstanding issues
    - [x] Drop-down fields render improperly
    - [x] Input field widths are incorrect on creation of a block / field
    - [x] Lint
- [x] Update "merge" documentation / clean-up script
- [x] Testing
    - [x] VM playground
    - [x] GUI playground
- [x] Squash before merging

### CODE REVIEW

**Comment by rachel-fenichel** [comment]
You still need to remove blockly_compressed, blocks_compressed, etc.  Also msg/json and msg/js files, since they don't all seem to be cleaned up.

**Comment by thisandagain** [comment]
@rachel-fenichel @picklesrus @rschamp This is ready for review. 🎉

**Comment by rschamp** [comment]
Testing in scratch-gui, if I load project 119615668 and click the Lists category, I receive an error:
```
TypeError: Cannot read property 'setAttribute' of undefined

Blockly.FieldDropdown.positionArrow (gui.js:37340)
Blockly.Field.render_ (gui.js:36902)
Blockly.Field.init (gui.js:36897)
Blockly.FieldTextInput.init (gui.js:37293)
Blockly.FieldTextDropdown.init (gui.js:37355)
Blockly.Input.init (gui.js:36924)
Blockly.BlockSvg.initSvg (gui.js:37187)
Blockly.Xml.domToBlock (gui.js:37035)
Blockly.Flyout.show (gui.js:37417)
Blockly.Toolbox.setSelectedItem (gui.js:37476)
(anonymous) (gui.js:37477)
e (gui.js:37550)
```

This doesn't happen on the previous build of scratch-gui.

**Comment by rachel-fenichel** [comment]
Doesn't matter what project.  Field render is being called earlier than it used to be.  FieldDropdown's positionArrow function needs to start with 
```
if (!this.arrow_) {
    return 0;
  }
```
to avoid trying to position something that doesn't exist.

**Review by rschamp — APPROVED** [review]
👍  I haven't been able to cause any other weirdness in the vm or gui playgrounds.

File: core/block_svg.js [hunk 1]
```diff
@@ -977,12 +1029,14 @@ Blockly.BlockSvg.prototype.onMouseMove_ = function(e) {
       // Switch to unrestricted dragging.
       Blockly.dragMode_ = Blockly.DRAG_FREE;
       Blockly.longStop_();
+      
+      // IS THIS COMMENT RELEVANT STILL???
```
Comment (picklesrus) [inline core/block_svg.js]: This is from one of my PRs to your local branch that I forgot to resolve. Would you mind removing it?

I believe the comment below is also irrelevant now too.  At least, as far as I can tell insertion markers/connection highlighting are working okay.  I remember having a conversation with Tim about this and we couldn't break it either... @rachel-fenichel: do you have any memory of what this means?

File: core/flyout_horizontal.js [hunk 2]
```diff
@@ -94,7 +94,8 @@ Blockly.HorizontalFlyout.prototype.getMetrics_ = function() {
   }
   var viewHeight = this.height_;
   if (this.toolboxPosition_ == Blockly.TOOLBOX_AT_TOP) {
-    viewHeight += this.MARGIN - this.SCROLLBAR_PADDING;
+    viewHeight += this.MARGIN;
+//    viewHeight += this.MARGIN - this.SCROLLBAR_PADDING;
```
Comment (picklesrus) [inline core/flyout_horizontal.js]: This line can go too

**Review by picklesrus — COMMENTED** [review]
Two minor comments, but otherwise looks good.

**Comment by thisandagain** [comment]
@rachel-fenichel @picklesrus Ok! Comments addressed.

@rschamp Good catch! I'm wondering if we should resolve that now or later? I tried doing a quick look and can't quite figure out why that exception is being thrown though it seems to be related to `field_numberdropdown` not inheriting from `field_number` properly.

**Comment by picklesrus** [comment]
@thisandagain,
Did you mean this error: TypeError: Cannot read property 'setAttribute' of undefined
Blockly.FieldDropdown.positionArrow (gui.js:37340)?

Your recent commit (https://github.com/LLK/scratch-blocks/pull/776/commits/ad1c6b1791e32968e33053746fbfb3276774d82e) should have fixed that. Maybe it hadn't made it to scratch-gui when @rschamp was testing?

**Comment by thisandagain** [comment]
@picklesrus No. The error we see when you select the "List" category is:
```
Uncaught TypeError: this.setConstraints_ is not a function
    at Blockly.FieldNumberDropdown.Blockly.FieldNumber.getNumRestrictor (vertical.js:2849)
    at new Blockly.FieldNumberDropdown (vertical.js:2857)
    at Blockly.BlockSvg.Blockly.Block.interpolate_ (vertical.js:2639)
    at Blockly.BlockSvg.Blockly.Block.jsonInit (vertical.js:2635)
    at Blockly.BlockSvg.init (vertical.js:1250)
    at Blockly.BlockSvg.Blockly.Block (vertical.js:2606)
    at new Blockly.BlockSvg (vertical.js:2662)
    at Blockly.WorkspaceSvg.newBlock (vertical.js:2544)
    at Object.Blockly.Xml.domToBlockHeadless_ (vertical.js:2519)
    at Object.Blockly.Xml.domToBlockHeadless_ (vertical.js:2521)
```

**Comment by rachel-fenichel** [comment]
@thisandagain interesting--I don't see that on the vertical playground.  Is that specific to a particular project?

**Comment by thisandagain** [comment]
@rachel-fenichel Ah. Yes ... nevermind, that does not appear to be generalized. I think we are good.

**Pull request merged**